### PR TITLE
Add nil checks to SecretStoreLocked

### PR DESCRIPTION
**Pull request by patrickxb**
r? @maxtaco

### CODE REVIEW

File: go/libkb/secret_store.go [hunk 1]
```diff
@@ -125,24 +125,36 @@ func NewSecretStoreLocked(g *GlobalContext) *SecretStoreLocked {
 }
 
 func (s *SecretStoreLocked) RetrieveSecret(username NormalizedUsername) ([]byte, error) {
+	if s == nil || s.SecretStoreAll == nil {
```
Comment (maxtaco) [inline go/libkb/secret_store.go]: Do you know where the typed `nil` is coming from?

File: go/libkb/secret_store.go [hunk 2]
```diff
@@ -125,24 +125,36 @@ func NewSecretStoreLocked(g *GlobalContext) *SecretStoreLocked {
...
+		return nil, nil
+	}
 	s.Lock()
 	defer s.Unlock()
 	return s.SecretStoreAll.RetrieveSecret(username)
 }
 
 func (s *SecretStoreLocked) StoreSecret(username NormalizedUsername, secret []byte) error {
+	if s == nil || s.SecretStoreAll == nil {
+		return nil
```
Comment (maxtaco) [inline go/libkb/secret_store.go]: Should this be an error, to call `StoreSecret` on a `nil` secret store?

Comment (patrickxb) [inline go/libkb/secret_store.go]: This is all temporary...once the SecretStoreFile branch makes it to master, it will never be nil.

**Comment by maxtaco** [comment]
OK 👍

**Pull request merged**